tests/unit/Snak/PropertyValueSnakTest.php

```diff
@@ -121,15 +121,62 @@ public function notEqualsProvider() {
 		];
 	}
 
-	public function testSerialize() {
-		$snak = new PropertyValueSnak( new PropertyId( 'P1' ), new StringValue( 'a' ) );
-		$this->assertSame( 'a:2:{i:0;i:1;i:1;C:22:"DataValues\StringValue":1:{a}}', $snak->serialize() );
+	public function provideDataToSerialize() {
+		$p2 = new PropertyId( 'P2' );
+		$p2foo = new PropertyId( 'foo:P2' );
+		$value = new StringValue( 'b' );
+
+		return [
+			'string' => [
+				'a:2:{i:0;s:2:"P2";i:1;C:22:"DataValues\StringValue":1:{b}}',
+				new PropertyValueSnak( $p2, $value ),
+			],
+			'foreign' => [
+				'a:2:{i:0;s:6:"foo:P2";i:1;C:22:"DataValues\StringValue":1:{b}}',
+				new PropertyValueSnak( $p2foo, $value ),
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider provideDataToSerialize
+	 */
+	public function testSerialize( $expected, Snak $snak ) {
+		$serialized = $snak->serialize();
+		$this->assertSame( $expected, $serialized );
+
+		$snak2 = new PropertyValueSnak( new PropertyId( 'P1' ), new StringValue( 'a' ) );
+		$snak2->unserialize( $serialized );
+		$this->assertTrue( $snak->equals( $snak2 ), 'round trip' );
 	}
 
-	public function testUnserialize() {
+	public function provideDataToUnserialize() {
+		$p2 = new PropertyId( 'P2' );
+		$p2foo = new PropertyId( 'foo:P2' );
+		$value = new StringValue( 'b' );
+
+		return [
+			'legacy' => [
+				new PropertyValueSnak( $p2, $value ),
+				'a:2:{i:0;i:2;i:1;C:22:"DataValues\StringValue":1:{b}}'
```

Comment:
It does not affect test result but I believe to be 100% accurate "i:2" here should become "i:1". Minor thing, I am going to fix this, and squash into other commits.